mshadow/dot_engine-inl.h

```diff
@@ -614,20 +618,20 @@ struct BLASEngine<gpu, float> {
   }
   inline static void gemm(Stream<gpu> *stream,
                           bool transa, bool transb,
-                          int m, int n, int k, float alpha,
-                          const float *A, int lda,
-                          const float *B, int ldb, float beta,
-                          float *C, int ldc) {
+                          index_t m, index_t n, index_t k, float alpha,
+                          const float *A, index_t lda,
+                          const float *B, index_t ldb, float beta,
+                          float *C, index_t ldc) {
     cublasStatus_t err = cublasSgemm(Stream<gpu>::GetBlasHandle(stream),
```

Comment:
Does lda, ldb, ldc support int64 in cublas?
```
cublasStatus_t cublasSgemm(cublasHandle_t handle,
                           cublasOperation_t transa, cublasOperation_t transb,
                           int m, int n, int k,
                           const float           *alpha,
                           const float           *A, int lda,
                           const float           *B, int ldb,
                           const float           *beta,
                           float           *C, int ldc)

```

Reply:
That's why I want to have a dot/GEMM unit test for large input tensors. Same problem may also exist for openblas and MKL.